src/content/blog/astro-4-beta.mdx

```diff
@@ -0,0 +1,62 @@
+---
+title: "Astro 4.0 Beta"
+description: "Astro 4.0 beta is released."
+publishDate: "November 22, 2023"
+authors:
+  - matthew
+coverImage: "/src/content/blog/_images/astro-4beta/post-header-4.0beta.webp"
+socialImage: "/src/content/blog/_images/astro-4beta/post-header-4.0beta.webp"
+lang: "en"
+---
+
+Today we've released the first beta of Astro 4! Those feeling adventurous can install the new beta, review our in-progress upgrade docs, and report any bugs you find to the [Astro discord](https://astro.build/chat).
+
+## What to expect in Astro 4
+
+We're releasing Astro 4 to stay up to date with the latest version of Vite, which just released Vite 5 last month.
+
+As Vite 5 is *also* a small major, most projects will be able to upgrade to Astro 4 with only minimal changes to your code, if any at all.
+
+Additionally, in Astro 4 we are planning to bring some of the new features we have been working on over the last few months out of experimental status and mark them as stable. We are still deciding *which* features those will be. Stay tuned for more!
+
+## Trying out the beta
+
+You can try out the new beta in your existing app by installing the `astro@beta` package with your favorite package manager.
+
+```sh
+npm install astro@beta
+pnpm install astro@beta
+yarn install astro@beta
+```
+
+If starting a new project you can also try out the beta by running:
+
+```sh
+npm init astro@beta
+```
+
+Again, substitute with your favorite package manager. We have not completed the upgrade guide yet, but you can already find the most significant changes in the [Astro 4.0 upgrade guide pull request](https://github.com/withastro/docs/pull/5481). You can also follow its related issue which is tracking all breaking changes.
+
+## Astro 3.6
+
+Additionally we've recently released Astro 3.6 which comes with new events in the `<ViewTransitions />` router as well as fallback for `prefetch` in reduced data scenarios.
+
+### View Transition events
+
+Astro 3.6 adds these three new events as part of the `<ViewTransitions />` router's lifecycle:
+
+`astro:before-preparation` runs immediately before the *preparation* step, where the content of the new page is loaded. This can be used to override the behavior of loading the new page, for example to show a loading spinner.
```

Comment:
```suggestion
`astro:before-preparation` runs immediately before the *preparation* step, where the content of the new page is loaded. This can be used to override the behavior of loading the new page, for example, to show a loading spinner.
```